.ci/pitest.sh

```diff
@@ -67,6 +67,14 @@ pitest-header)
   checkPitestReport "${ignoredItems[@]}"
   ;;
 
+pitest-sizes)
+  mvn --no-transfer-progress -e -P$1 clean test-compile org.pitest:pitest-maven:mutationCoverage;
+  declare -a ignoredItems=(
+  "MethodLengthCheck.java.html:<td class='covered'><pre><span  class='survived'>                &#38;&#38; isExceedingLimit(length)) {</span></pre></td></tr>"
```

Comment:
We do a lot to reduce list of such ignore items.
Please share challenges that force to do this.

We would rather has pure Junit to kill mutation to avoid ignored from pitest

Reply:
I added this check to make sure that we really need to iterate over whole AST to count exact number of lines.
For example, you have config like
```
MethodLengthCheck
countEmpty = false
max = 1000
```
and you have class with 100 methods, each of them is less than 1000 lines and have 200 elements in ast. Apparently, check will produce 0 violations for such class.
And now we can see difference that this "redundant" check gives us:
1. Without it, you will iterate over every node of ast of each method, plus create 100 new boolean arrays and 100 array queues. Basically, for such cases in will result in `O(n + m)` memory complexity and `O(n)` time complexity, where `n` is number of ast nodes in all methods and `m` is number of lines. Plus, all those arrays has to be collected by gc, which adds some penalty.

2. You just make a shortcut check for raw length, thus skip iteration. That will result in O(1) in both space and time complexity.

I have no idea how to explain all this to pitest. Of course, parsing takes a lot more time than inspection itself and show it by single input is impossible. For such cases, difference can be shown on big scale.

Reply:
ok, I removed it. I hope someday we will have tests that can prove not only logic, but also space/time advantages.